File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.py
def solution(t, p):
    answer = 0
    a = []
    n = len(p)
    for i in range(len(t)-n+1):
        a.append(t[i:i+n])
    for i in a:
        if i <= p:
            answer += 1

    return answer

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
import java.util.Arrays;

class Solution {
    public int solution(String t, String p) {
        int n = p.length();
        long pNum = Long.parseLong(p);
        int answer = 0;
        
        for(int i = 0; i < t.length()-n+1; i++){
            String temp = t.substring(i, i+n);
            if (Long.parseLong(temp) <= pNum){
                answer++;
            }
        }
        
        return answer;
    }
}